gcsio/src/test/java/com/google/cloud/hadoop/gcsio/PerformanceCachingGoogleCloudStorageTest.java

```diff
@@ -236,6 +236,81 @@ public void testListObjectInfo_delimiter() throws IOException {
     assertThat(cache.getAllItemsRaw()).containsExactlyElementsIn(expectedCached);
   }
 
+  @Test
+  public void testListObjectNames_prefixDir_delimiter() throws IOException {
+    String prefixADir = PREFIX_A + "/";
+
+    String prefixABADir = PREFIX_ABA.substring(0, PREFIX_ABA.lastIndexOf('/') + 1);
+    GoogleCloudStorageItemInfo itemABAPrefix = createInferredDirectory(BUCKET_A, prefixABADir);
+
+    List<String> expectedResult =
+        Lists.newArrayList(ITEM_A_AA.getObjectName(), itemABAPrefix.getObjectName());
+    List<GoogleCloudStorageItemInfo> expectedCached = Lists.newArrayList(ITEM_A_AA, itemABAPrefix);
+
+    List<String> result = gcs.listObjectNames(BUCKET_A, prefixADir, ListObjectOptions.DEFAULT);
+
+    // Verify the delegate call.
+    verify(gcsDelegate)
+        .listObjectNames(eq(BUCKET_A), eq(prefixADir), eq(ListObjectOptions.DEFAULT));
+
+    // Verify the result.
```

Comment:
Why some of the comments are one line off?

Reply:
Done.